### ISSUE TITLE
Feature request: use neorg for built-in documentation

### ISSUE THREAD

**Issue by NTBBloodbath** [NTBBloodbath/doom-nvim#98]
It could be really cool if we use [neorg](https://github.com/vhyrro/neorg) for providing Doom's built-in documentation instead of the ugly built-in Vim docs format (`.txt`). Also Doom Emacs uses `org` mode for their docs and it looks really good so why can't we have beauty built-in docs too? :)

Maybe not today, maybe not tomorrow, but we will surely do this change after neorg reaches version 0.1

> Current Vim `.txt` docs

![Vim docs demo](https://user-images.githubusercontent.com/36456999/129492749-c7adea45-90a5-437d-a4e8-d50ba0c18654.png)

> Neorg docs

![Neorg docs demo](https://user-images.githubusercontent.com/36456999/129492789-5b577496-8d9c-4106-8d48-e9abfc16df0e.png)

**Comment by molleweide** [comment]
Yesssss!! I am trying to convert more and more of my own files to neorg

**Comment by NTBBloodbath** [comment]
So I finally included the Neorg-based documentation in the development branch (doom v3.2.0-beta).

We are using our own command (`:DoomManual`) and not the `:help` command due to some reasons that I'm gonna list here

> tl;dr since the Neovim help command is meant to work with the Neovim docs format (`.txt`) it doesn't work well with Neorg files so we implemented our own help command haha

- The `:help` command sets some useful options like disabling line numbers, but also overrides the norg filetype in the buffer and causes the tree-sitter to work in weird ways (almost all the syntax highlighting dies)
- The help tags (`:help doom` by example) cannot be automatically generated with `:helptags` command so they're not working properly and I don't know why because the format for the tags file is correct.

This is why we have the `:DoomManual` command, we manually set the useful options for the user manual buffer without killing the tree-sitter parser stuff and also the command will always place the user cursor under the Table of Contents section, where the user can easily press `<cr>` to jump to internal document sections or even to external links like github repositories. In that way we can also cover the tags stuff with even more power!
